odoo/http.py

```diff
@@ -1090,6 +1142,11 @@ def default_lang(self):
         except (ValueError, KeyError):
             return DEFAULT_LANG
 
+    def _geoip_resolve(self):
+        if not (root.geoip_resolver and self.httprequest.remote_addr):
+            return {}
+        return root.geoip_resolver.resolve(self.httprequest.remote_addr) or {}
```

Comment:
Maybe this would be cleaner if `geoip_resolver` was a "dummy" resolver (always returning an empty dict) in the case where `GeoIPResolver.open` returns `None`?

Reply:
I agree, I decided to keep the same API as before, I should had made it clean. I'm doing it in another PR, I'm on a tight schedule here